src/components/course/data/hooks.jsx

```diff
@@ -583,3 +585,55 @@ export function useUserHasSubsidyRequestForCourse(courseKey) {
     requestsBySubsidyType,
   ]);
 }
+
+const checkRedemptionEligiblity = async ({
+  lmsUserId,
+  courseRunKeys,
+}) => {
+  const courseService = new CourseService();
+  const canRedeemResponse = await courseService.fetchCanRedeem({
+    lmsUserId,
+    courseRunKeys,
+  });
+  const canRedeemByCourseRun = canRedeemResponse.map((canRedeemForCourseRun) => {
+    const {
+      redemption,
+      subsidyAccessPolicy,
+    } = canRedeemForCourseRun;
+    const resultForCourseRun = {
+      courseRunKey: canRedeemForCourseRun.courseRunKey,
+      canRedeem: false,
+      alreadyRedeemed: false,
+      redemption: null,
+      redeemablePolicy: null,
+    };
+    if (redemption?.state === 'committed') {
+      resultForCourseRun.alreadyRedeemed = true;
+      resultForCourseRun.redemption = redemption;
+    }
+    if (subsidyAccessPolicy) {
+      resultForCourseRun.canRedeem = true;
+      resultForCourseRun.redeemablePolicy = subsidyAccessPolicy;
+    }
+    return resultForCourseRun;
+  });
+  return canRedeemByCourseRun;
+};
+
+/**
+ * TODO
+ * @param {*} param0
+ * @returns
+ */
+export const useCheckAccessPolicyRedemptionEligibility = ({
+  courseRunKeys,
+}) => {
+  const { id: lmsUserId } = getAuthenticatedUser();
+  const isFeatureEnabled = getConfig().FEATURE_ENABLE_EMET_REDEMPTION;
+
+  return useQuery({
```

Comment:
Remember to invalidate the course run key given a user has redeemed a specific course run to invalidate the cached data 

OR

disable caching for now

Reply:
I added some logic to invalidate the `can-redeem` query cache upon successful redemption via the `handleRedeemSuccess` function returned by `useRedemptionStatus` ([docs](https://tanstack.com/query/v4/docs/react/guides/query-invalidation)):

```jsx
const handleRedeemSuccess = (transaction) => {
  queryClient.invalidateQueries({ queryKey: ['can-user-redeem-course'] });
  setRedemptionStatus('success');

  // redirect to courseware
  const { coursewareUrl } = transaction;
  global.location.assign(coursewareUrl);
};
```